### PR TITLE
bpo-32136: Separate embedding tests from C API tests

### PR DESCRIPTION
**Pull request by ncoghlan**
Some parts of the C API are only relevant to larger
applications embedding CPython as a runtime engine.

The helpers to test those APIs are already separated
out into Programs/_testembed.c, this update moves
the associated test cases out into their own dedicated
test file.

https://bugs.python.org/issue32136

### CODE REVIEW

File: Lib/test/test_embed.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
+# Run the tests in Programs/_testembed.c (tests for the CPython embedding APIs)
+from test import support
+import unittest
+
+from collections import namedtuple
+import os
+import re
+import subprocess
+import sys
+
```
Comment (vstinner) [inline Lib/test/test_embed.py:10]: Would you mind to add an empty line for PEP8, please? :-)

File: Lib/test/test_capi.py [hunk 2]
```diff
@@ -420,190 +420,6 @@ def test(self):
         self.assertEqual(_testcapi.argparsing("Hello", "World"), 1)
```
Comment (vstinner) [inline Lib/test/test_capi.py:421]: pyflakes found 2 unused imports:

Lib/test/test_capi.py:4:1: F401 'collections.namedtuple' imported but unused
Lib/test/test_capi.py:7:1: F401 'platform' imported but unused

File: Lib/test/test_embed.py (hunk 1 above)
Comment (ncoghlan) [reply Lib/test/test_embed.py:10]: Added!

File: Lib/test/test_capi.py (hunk 2 above)
Comment (ncoghlan) [reply Lib/test/test_capi.py:421]: Both removed now.

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**